woocommerce/woocommerce-ios
# Account Creation: Navigate to custom login UI for existing account

Part of #10590 


## Description

Previously, we were integrating the WPAuthenticator library for handling login when a user starts the account creation flow with an existing email address. This adds an additional step that requires them to confirm their email address.

This PR updates the login flow from account creation to use the custom UI that we created during the Jetpack setup project. The goal is to display the password screen (or magic link screen for passwordless accounts) immediately after the user enters an existing email address.

Changes include:
- Created a new coordinator `WPComLoginCoordinator` with the custom UI created during the Jetpack setup project. This coordinator handles checking WPCom account with an input email address and present subsequent steps for the login flow.
- Updated the `AccountCreationFormViewModel` with a separate async callback when an existing email address is submitted.
- Updated the `LoggedOutStoreCreationCoordinator` to use the new login coordinator when an existing email address is submitted.
- Updates to the existing custom login UI: password, magic link, 2FA screens to show/hide the Jetpack header and customize the title labels.
- Moved the WPCom login related files to the Authentication group.

📝 Notes:
- Due to the size of this PR, unit tests for `WPComLoginCoordinator` will be added in a separate PR.
- A few other to-dos:
  - Security key support.
  - Magic link flow cannot navigate to store creation automatically. I haven't yet found a solution for that.
  - Updates for `JetpackSetupCoordinator` to use `WPComLoginCoordinator` and remove the duplicated code.

## Testing instructions

- Log out of the app if needed.
- On the prologue screen, select Create a new store.
- On the account creation screen, enter the email address of an existing WPCom account.
- Notice that the custom password screen is displayed immediately. If your account is passwordless (e.g created with a social account), the magic link screen is displayed immediately.
- Proceed to enter your password and 2FA code if required.
- After the login succeeds, you should be navigated to the store picker screen with the store creation modal presented. If you log in with a magic link, the store creation modal will not be presented automatically.

Please feel free to test the login through the Jetpack setup flow again following the steps in #9113 to make sure that everything still works properly.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/26dd92b1-e055-4891-9b68-2d21166cc38c



---
